api/nodes/permissions.py

```diff
@@ -63,6 +65,51 @@ def has_object_permission(self, request, view, obj):
             has_auth = parent_node.can_edit(auth)
             return has_auth
 
+
+class CanCommentOrPublic(permissions.BasePermission):
```

Comment:
It seems more appropriate to put these new classes in `api/comments/permissions.py`.